src/pools/D3MAaveV3TypePool.sol

```diff
@@ -0,0 +1,292 @@
+        PoolLike.ReserveData memory data = pool.getReserveData(dai_);
+        require(data.aTokenAddress               != address(0), "D3MAaveV3TypePool/invalid-adai");
+        require(data.stableDebtTokenAddress      != address(0), "D3MAaveV3TypePool/invalid-stableDebt");
+        require(data.variableDebtTokenAddress    != address(0), "D3MAaveV3TypePool/invalid-variableDebt");
+
+        adai = ATokenLike(data.aTokenAddress);
+        stableDebt = ATokenLike(data.stableDebtTokenAddress);
+        variableDebt = ATokenLike(data.variableDebtTokenAddress);
+
+        dai.approve(pool_, type(uint256).max);
+
+        hub = hub_;
+        vat = VatLike(D3mHubLike(hub_).vat());
+        vat.hope(hub_);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    modifier auth {
+        require(wards[msg.sender] == 1, "D3MAaveV3TypePool/not-authorized");
+        _;
+    }
+
+    modifier onlyHub {
+        require(msg.sender == hub, "D3MAaveV3TypePool/only-hub");
+        _;
+    }
+
+    // --- Math ---
+    uint256 internal constant RAY = 10 ** 27;
+    function _rdiv(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = (x * RAY) / y;
+    }
+    function _min(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = x <= y ? x : y;
+    }
+    function _divup(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        unchecked {
+            z = x != 0 ? ((x - 1) / y) + 1 : 0;
+        }
+    }
+
+    // --- Admin ---
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    function file(bytes32 what, address data) external auth {
+        require(vat.live() == 1, "D3MAaveV3TypePool/no-file-during-shutdown");
+        if (what == "hub") {
+            vat.nope(hub);
+            hub = data;
+            vat.hope(data);
+        } else if (what == "king") king = data;
+        else revert("D3MAaveV3TypePool/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    // Deposits Dai to Aave in exchange for adai which is received by this contract
+    // Aave: https://docs.aave.com/developers/v/2.0/the-core-protocol/lendingpool#deposit
+    function deposit(uint256 wad) external override onlyHub {
+        uint256 scaledPrev = adai.scaledBalanceOf(address(this));
+
+        pool.deposit(address(dai), wad, address(this), 0);
+
+        // Verify the correct amount of adai shows up
+        uint256 interestIndex = pool.getReserveNormalizedIncome(address(dai));
+        uint256 scaledAmount = _rdiv(wad, interestIndex);
+        require(adai.scaledBalanceOf(address(this)) >= (scaledPrev + scaledAmount), "D3MAaveV3TypePool/incorrect-adai-balance-received");
+    }
+
+    // Withdraws Dai from Aave in exchange for adai
+    // Aave: https://docs.aave.com/developers/v/2.0/the-core-protocol/lendingpool#withdraw
```

Comment:
this should point to the v3 docs

Reply:
Fixed https://github.com/makerdao/dss-direct-deposit/pull/102/commits/5eb5703a0abc361073e4eabe4e111cf31863e37b